File: Frontend/src/app/DTOs/Create/CreateUpsertSetQuestionsDTO.ts
import { CreateQuestionDistributeDTO } from "./createQuestionDistributeDTO";
import { CreateQuestionMathematicDTO } from "./createQuestionMathematicDTO";
import { CreateQuestionMultipleChoiceDTO } from "./createQuestionMultipleChoiceDTO";
import { CreateQuestionOpenQuestionDTO } from "./createQuestionOpenQuestionDTO";
import { CreateQuestionTextFieldDTO } from "./createQuestionTextFieldDTO";
import { CreateQuestionTrueFalseDTO } from "./createQuestionTrueFalseDTO";
import { CreateQuestionWordDTO } from "./createQuestionWordDTO";

export interface CreateUpsertSetQuestionsDTO {
    setId: string;
    questionsDistribute: CreateQuestionDistributeDTO[];
    questionsMathematic: CreateQuestionMathematicDTO[];
    questionsMultipleChoice: CreateQuestionMultipleChoiceDTO[];
    questionsOpenQuestion: CreateQuestionOpenQuestionDTO[];
    questionsTextField: CreateQuestionTextFieldDTO[];
    questionsTrueFalse: CreateQuestionTrueFalseDTO[];
    questionsWord: CreateQuestionWordDTO[];
}

File: Backend/DTOs/Create/CreateUpsertSetQuestionsDTO.cs
namespace Learnz.DTOs;
public class CreateUpsertSetQuestionsDTO
{
    public Guid SetId { get; set; }
    public List<CreateQuestionDistributeDTO> QuestionsDistribute { get; set; }
    public List<CreateQuestionMathematicDTO> QuestionsMathematic { get; set; }
    public List<CreateQuestionMultipleChoiceDTO> QuestionsMultipleChoice { get; set; }
    public List<CreateQuestionOpenQuestionDTO> QuestionsOpenQuestion { get; set; }
    public List<CreateQuestionTextFieldDTO> QuestionsTextField { get; set; }
    public List<CreateQuestionTrueFalseDTO> QuestionsTrueFalse { get; set; }
    public List<CreateQuestionWordDTO> QuestionsWord { get; set; }
}